Update template for the spherex class

jonathansick opened the pull request:
- [x] Convert tex template to use spherex class
- [x] use correct handle format and separate number from handle for more reliability.
- [x] diagram index is no longer part of lander.yaml because it is included in the document now.
- [x] Document build workflow in GitHub Actions
- [x] Store PDF artifact in GitHub Actions
- [x] Release workflow that also stores the PDF asset.

gpdf commented:
Some of the existing Rubin templates, e.g., `test_report`, have an instantiated test document in the repo.  This has pluses and minuses - it can get out of date with the template, if the workflow isn't carefully designed, but it also makes the rendered state of the template more apparent.

Example:
https://github.com/lsst/templates/tree/master/project_templates/test_report

Any thoughts about that for this?

gpdf commented:
I kind of like the Rubin notion of file templates and project templates.  If you are supportive of replicating that approach, I'd be happy to go with it from the start.  It won't be long before we start needing analogs to the things Rubin has in file_templates.

gpdf commented on pipeline-module/cookiecutter.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "handle": "SPHEREX-P-001",
+  "handle": "SSDC-MS-000",
```
I think if this is a dedicated template for module specifications, we might only give the cookiecutter user the serial number to input rather than the whole handle.

Cookiecutter would let us then confirm the generated handle, right, just so the user would get confidence from seeing it?

jonathansick replied:
That was my original plan, but I ran into issues doing the zero-padded string formatting within the context of the Jinja template inside cookiecutter.json. This `handle` variable is a work-around to get something working quickly. I think I'll need to write a small Jinja extension to do the string formatting.

gpdf replied:
What if we just reminded the user to type the leading zeros, for now?  This worked for Roman, the way you set up that template.

gpdf reviewed (changes requested):
I think this is ready to roll out once the change to how the handle is calculated is included.  There might be tweaks after that.

jonathansick commented:
If you want to replicate the Rubin templates repository set up, we can do that. This repository is sticking closely to Cookiecutter, while for the Rubin templates repository, we introduced our own management layer on top of Cookiecutter called templatekit (https://templatekit.lsst.io). Templatekit is what makes single-file templates possible, as well as makes the Slack integration possible (in conjunction with the [templatebot](https://github.com/lsst-sqre/templatebot) and [lsst-templatebot-aide](https://github.com/lsst-sqre/lsst-templatebot-aide) services). Templatekit also provides the SCons extensions that power the in-repository previews.

There have been some drawbacks to the templatekit approach. Some contributors finding it inconvenient to regenerate the examples. I think that can be solved with a pre-commit plugin that re-runs the scons commands before commit.

The good news is that templatekit is already generic in its design and can be applied here easily. Let me know if you'd like that done. I'd probably want to do that in a subsequent PR to this one to keep the diffs simple.

gpdf commented:
I think I'd need you to tell me more about the drawbacks you mentioned before converting to `templatekit`.

gpdf commented:
Some comments from today's meeting:

1. We'd like to drop the "status" altogether; it's vestigial now that all the modules have an initial document version available.
2. We'd like to move the "difficulty" into the rendered view of the document.  I think that means it needs a macro in the class file and an appropriate entry in the template?  And perhaps moves it out of the `lander.yaml` file?

gpdf commented:
I've pushed removing "status" as well as a slight rearrangement of the .tex template.

jonathansick commented:
I've added `\difficulty` and switched the handle command to `\spherexHandle`. These changes rely on updates to spherex-tex in https://github.com/SPHEREx/spherex-tex/pull/3

jonathansick commented:
Also, do you wish to have a CODEOWNERS file set up for these MS documents?

gpdf commented on pipeline-module/{{cookiecutter.handle|lower|slugify}}/.gitignore:
```diff
@@ -1,5 +1,5 @@
 # Published by CI
-{{ cookiecutter.handle|lower|slugify }}.pdf
+{{ cookiecutter.handle }}.pdf
```
Why no lowercase, etc. any more?

jonathansick replied:
The PDF file is the uppercase handle, so that's why the ignore is for the uppercase handle as well. The repo itself is still lowercase.

gpdf reviewed (commented):
Looks good.  Just one question in a comment.
